desktop/libs/aws/src/aws/s3/s3connection.py

```diff
@@ -73,13 +74,17 @@ def __init__(self, username, aws_access_key_id=None, aws_secret_access_key=None,
     # No auth handler with RAZ
     anon = RAZ.IS_ENABLED.get() and not IS_SELF_SIGNING_ENABLED.get()
 
+    # Disable instance metadata fetching when RAZ is enabled
+    raz_s3_provider = Provider(provider, aws_access_key_id, aws_secret_access_key, security_token, profile_name)
+    raz_s3_provider.MetadataServiceSupport['aws'] = False
```

Comment:
Do we need to disable MetadataServiceSupport for all cases or just when RAZ is in use?

```suggestion
    raz_s3_provider.MetadataServiceSupport['aws'] = not RAZ.IS_ENABLED.get()
```